### ISSUE TITLE
Project: Example Notebooks

### ISSUE THREAD

**Issue by samagra14** [aimacode/aima-java#441]
This is the discussion issue for Example Notebooks project.  The **notebook section** is completely new. Only a few chapters have been implemented in the form of notebooks. The general approach to any notebook is as follows:
* Explain briefly the **ideas from the book**.
* Explain the API by implementing the interfaces for some custom approach to the explained ideas. For eg: extend the `Problem` class to implement some **custom problem**.
* Solve the same problem using already implemented APIs from the code and **compare the merits** of using the APIs from the repository if such APIs exist.
* Apply algorithms on **various problems** related to the domain.
* There are a few sections from the book( for eg: Dimensionality Reduction or Ontological Engineering) where the authors have explained the **concept without providing a formal algorithm**. Such topics must be covered in the notebooks

Before starting to work on a new notebook:
1. Open a new issue with the following heading: **Notebook: Chapter Name - Version **. Check that the issue is not assigned to anyone.
2. Mention a topics list of what you will be implementing in the notebook for that particular chapter.
3. Mention @samagra14 to discuss what can go in that particular notebook.

**Comment by jsuyash1514** [comment]
@samagra14 I would like to work on this project. Should I proceed?

**Comment by samagra14** [comment]
Yes, sure. Open the issue for a chapter on which you want to work and keep mentioning the things that you are going to add so that we are on the same page regarding the contents of the notebook.

**Comment by ho-dor** [comment]
@samagra14 I am still not clear on how to contribute , can you please guide me in steps on how to look for issues and how to solve them ?

**Comment by adityaa30** [comment]
@samagra14 I want to work on Chp 7 : Logic agents. Shall i open an issue ?

**Comment by samagra14** [comment]
Hi @ho-dor 
These notebooks are support materials on how to use the code repository and how to apply various concepts from the book. So, the basic procedure is that you first read the chapter, then read the relevant module. Now, assume that you are a teacher and you want to explain to a new kid what the code is all about.
In other words, **_if I have a practical exam tomorrow of the given chapter and I don't have enough time, then it would be awesome for me that if I come to the notebook and get a thorough reading, then I should be able to perform fairly in the exam._**
